WIMP Frontend

Prasanthi-Peram opened the pull request.

Prasanthi-Peram committed:
create basic layout search bar and render a table

Prasanthi-Peram committed:
Minor fixes

Prasanthi-Peram committed:
Updated the code to React

Prasanthi-Peram committed:
Fuzzy search implemented

Prasanthi-Peram committed:
Footer updated

Prasanthi-Peram committed:
Styling added

rohan-b-84 commented on frontend/src/components/Footer.tsx:
```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import "./footer.css";
```
```suggestion

```

rohan-b-84 commented on frontend/index.html:
```diff
@@ -0,0 +1,13 @@
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <link rel="icon" type="image/x-icon" href="./src/assets/favicon.ico"  >
+    <title>WIMP-MetaKGP</title>
```
```suggestion
    <title>WIMP | MetaKGP</title>
```

rohan-b-84 commented on frontend/src/components/body.tsx:
```diff
@@ -0,0 +1,41 @@
+    <>
+      <div className="main">
+        <div
+          style={{
+            display: "flex",
+            flexDirection: "column",
+            alignItems: "center",
+          }}
```
add this inside .css file
btw, multiple css files not needed in this case, just import one css file.

rohan-b-84 commented on frontend/src/components/body.tsx:
```diff
@@ -0,0 +1,41 @@
+          <h1>Where Is My Prof?</h1>
+          <img
+            src={doodleImage}
+            style={{ marginTop: "-48px", width: "40%" }}
```
add in css file

rohan-b-84 commented on frontend/src/components/body.tsx:
```diff
@@ -0,0 +1,41 @@
+          <img
+            src={logo}
+            alt="MetaKGPlogo"
+            style={{ marginBottom: "-70px" }}
```
add in css file

rohan-b-84 commented on frontend/src/components/body.tsx:
```diff
@@ -0,0 +1,41 @@
+          />
+        </div>
+        <div>
+          <p style={{ margin: "0 auto", textAlign: "left", width: "70%" }}>
```
add in css file

rohan-b-84 commented on frontend/src/components/table.tsx:
```diff
@@ -0,0 +1,136 @@
+              <a
+                href={`${selectedProfessor.website}`}
+                target="_blank"
+                style={{
+                  color: "inherit",
+                  textDecoration: "none",
+                  fontWeight: "lighter",
+                  fontSize: "56px",
+                }}
```
add in css file

rohan-b-84 commented on frontend/src/components/table.tsx:
```diff
@@ -0,0 +1,136 @@
+      {selectedProfessor && (
+        <div className="table-container">
+          <div className="table-caption">
+            <span style={{ color: "#007BFF" }}>
```
add in css file

rohan-b-84 commented on frontend/src/components/table.tsx:
```diff
@@ -0,0 +1,136 @@
+                {selectedProfessor.name}
+              </a>
+            </span>{" "}
+            <span style={{ fontWeight: "lighter", fontSize: "56px" }}>
```
add in css file

rohan-b-84 reviewed (changes requested):
Please make it responsive (Always think of making stuff responsive)

Margin and Padding is not the way to position stuff. Hard-coded values cannot be trusted as people might have different display dimensions or pixel ratio, relative values can't be trusted either for similar reasons. 

Also check the slack thread.

Prasanthi-Peram committed:
footer error fix

Prasanthi-Peram committed:
Styling updated

Prasanthi-Peram committed:
overflow fix

Dami-18 committed:
update css

Dami-18 committed:
add horizontal scroll in table

Faizan2005 committed:
Integrate iitkgp-erp-login package and resolve faculty department data fetching issue

Faizan2005 committed:
Fix HTML parsing for professor information and timetable extraction

- Updated HTML parsing logic to accurately extract professor details.
- Created  file to store headers and payload configurations.
- Moved  file to .

Faizan2005 committed:
Added .session to .gitignore

Faizan2005 committed:
Fix HTML parsing for professor information and timetable extraction

- Updated HTML parsing logic to accurately extract professor details.
- Created  file to store headers and payload configurations.
- Moved  file to config.pyto:              cannot open `to' (No such file or directory)
store:           cannot open `store' (No such file or directory)
headers:         cannot open `headers' (No such file or directory)
and:             cannot open `and' (No such file or directory)
payload:         cannot open `payload' (No such file or directory)
configurations.: cannot open `configurations.' (No such file or directory).sessionto: cannot open `to' (No such file or directory).gitignore'.

Faizan2005 committed:
Fix HTML parsing for professor information and timetable extraction

- Updated HTML parsing logic to accurately extract professor details.
- Created  file to store headers and payload configurations.
- Moved  file to .

Faizan2005 committed:
Remove unnecessary files, add comments

- Removed 'ter' file
- Added comments to
- Removed  file

harshkhandeparkar committed:
yeet: yeeted unused files

harshkhandeparkar committed:
feat: constants in a separate file

harshkhandeparkar committed:
feat: added slots map; formatted constants

harshkhandeparkar committed:
feat: added functions to fetch timetable and profs list

harshkhandeparkar committed:
feat: added function to parse prof's data

harshkhandeparkar committed:
feat: separate wimp module

harshkhandeparkar committed:
feat: updated and required requirements

harshkhandeparkar committed:
fix: fixed constants relative import

harshkhandeparkar committed:
feat: parse department timetable

harshkhandeparkar committed:
checkpoint: dept timetable and profs fetching complete

harshkhandeparkar committed:
feat: keep only unique professors

harshkhandeparkar committed:
feat: added most of the timetable parsing code

harshkhandeparkar committed:
refactor: moved timetable building code to a separate file

harshkhandeparkar committed:
feat: use bs4 to parse the anchor tag in faculty directory

harshkhandeparkar committed:
feat: completed handling of typos

harshkhandeparkar committed:
docs: added explanation for the typo issue

harshkhandeparkar committed:
fix: [WIP] updated the inaccuracy typeddict

harshkhandeparkar committed:
feat: added levenshtein to the requirements

harshkhandeparkar committed:
feat: updated data; removed unused data files

harshkhandeparkar committed:
feat: removed more unused files

harshkhandeparkar committed:
feat: removed even more unused files

harshkhandeparkar committed:
feat: added code to list all potential inaccuracies

harshkhandeparkar committed:
feat: formatted all files

harshkhandeparkar committed:
feat: moved to pnpm

harshkhandeparkar committed:
feat: added start script

harshkhandeparkar committed:
fix: removed unnecessary stuff

harshkhandeparkar committed:
feat: updated for wimp 2.0 data format

harshkhandeparkar committed:
feat: reduced header size

harshkhandeparkar committed:
fix: removed unused imports

harshkhandeparkar committed:
feat: slightly better data display. someone will fix the css i hope

harshkhandeparkar committed:
feat: added tooltip for course name

Prasanthi-Peram committed:
fix: modified table css

harshkhandeparkar committed:
yeet: yeet package log; use pnpm

harshkhandeparkar committed:
feat: autofocus for qol

harshkhandeparkar committed:
fix: use `, ` for room separator
